### PR TITLE
Add weapon ammo to the weapon name

### PR DESCRIPTION
**Pull request by Hirmuolio**
#### Summary

```SUMMARY: Features "Add weapon ammo to weapon name"```

#### Purpose of change

Quick quiz: What kind of ammo does RM216 SPIW use?

The game has so many guns, many of them are exotic or fictional with very non descriptive name. Having to check the item info to see what ammo they use is tiresome.

The solution is to add the used ammo to the weapon name.

#### Describe the solution

The default ammo is added to after the ammo count. If the weapon has no ammo count (no magazine) then the ammo is just at the end of the name.

Same is done to magazines. They will also have the ammo names on their name.

If the weapon/magazine is loaded with ammo then the type of the loaded ammo is used.

UPS usage is not added to the name.

The feature can be disabled/enabled in options. "Add ammo to weapon names" option in interface settings. Default enabled.

Screenshot:
ammo2

#### Describe alternatives you've considered
The ammo is not technically in the name of the imtem. So you can not search for it.
I considered placing the ammo in the name properly but that would push the ammo count off screen and that would not be even worse.

#### Testing
It doesn't seem to crash.
It works on weapons  that have had their ammo type changed by a gun mod.

### CODE REVIEW

**Comment by ymber** [comment]
I'd much prefer this being behind an option. There's a lot more text in the list with this and that makes it harder for me to find what I want at a glance.

**Comment by Night-Pryanik** [comment]
> Even if the weapon accepts other ammo types and is loaded with other ammo types it will still show default ammo.

This will be highly confusing and will cause all sorts of "Why does it shows X ammo when I loaded it with Y ammo?" questions. I'd prefer it showing the actual ammo loaded, and default ammo only when the gun is unloaded.

**Comment by Hirmuolio** [comment]
I am trying to make it show the current ammo if any is loaded.

There is some strange behaviour that I do not understand at all.

`ammotext = ammotype( ammo_current() )->name();`

This works only if the weapon/magazine are loaded with certain types of ammo. For example it works with "9x19mm JHP" but not with "9x19mm +P+". It gives this kind of error:

```
 DEBUG    : Tried to get invalid ammunition: 9mmP2

 FUNCTION : obj
 FILE     : d:\programming stuff\cataclysm-dda\src\ammo.cpp
 LINE     : 48
```

I have a feeling that I am doing something completely wrong and this shouldn't work at all.

**Comment by Hirmuolio** [comment]
Got something working. But I have a feeling this is a bad way to do this.

`item( ammo_current() ).ammo_type()->name();`

As far as I understand this gets the itype_id of currently loaded ammo, creates item from that itype_id, gets the ammotype of the created item and then gets the name of that ammotype.

File: src/item.cpp [hunk 1]
```diff
@@ -4024,16 +4024,31 @@ std::string item::display_name( unsigned int quantity ) const
         max_amount = to_joule( type->battery->max_capacity );
     }
 
+    std::string ammotext;
+    if( ( ( is_gun() && ammo_required() ) || is_magazine() ) && get_option<bool>( "AMMO_IN_NAMES" ) ) {
+        if( ammo_current() != "null" ) {
+            ammotext = item( ammo_current() ).ammo_type()->name();
```
Comment (BevapDin) [inline src/item.cpp]: See implementation of `item::ammo_type()`. Something like this should work:

```C++
ammotext = find_type( ammo_current() )->ammo->type->name()
           ^ gets an const itype*
                                      ^ accesses itype::ammo (islot_ammo or similar)
                                            ^ get the ammunition type id from the islot_ammo
```

Comment (Hirmuolio) [reply src/item.cpp]: This works perfectly.

**Comment by Hirmuolio** [comment]
I think this is pretty much ready.
Are there any weird  weapons/magazines that hold non-ammo items or no ammo at all? Or anything else weird that could cause problems?

File: src/options.cpp [hunk 2]
```diff
@@ -1549,6 +1549,10 @@ void options_manager::add_options_interface()
          translate_marker( "If true, show item symbols in inventory and pick up menu." ),
          false
        );
+    add( "AMMO_IN_NAMES", "interface", translate_marker( "Add ammo to weapon/magazine names" ),
+         translate_marker( "If true, the default ammo is added to weaon and magazine names. For example \"Mosin-Nagant M44 (4/5)\" becomes \"Mosin-Nagant M44 (4/5 7.62x54mm)\"." ),
```
Comment (Night-Pryanik) [inline src/options.cpp]: Weaon -> weapon.

Comment (anothersimulacrum) [reply src/options.cpp]:
```suggestion
         translate_marker( "If true, the default ammo is added to weapon and magazine names. For example \"Mosin-Nagant M44 (4/5)\" becomes \"Mosin-Nagant M44 (4/5 7.62x54mm)\"." ),
```

**Pull request merged**